### PR TITLE
Replace votingBalance -> derive.balances.all

### PR DESCRIPTION
**Pull request by kwingram25**
Remove defunct balances keys and update w new call

(Please check that calculations are accurate! @Tbaut @jacogr)

### CODE REVIEW

File: packages/api-derive/src/balances/index.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+export * from './all';
+export * from './allMulti';
```
Comment (jacogr) [inline packages/api-derive/src/balances/index.ts]: Not crazy about the `allMulti` naming here. :( However, I'm complaining without a better suggestion, so probably need to live with it.

File: packages/api-derive/src/balances/all.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { AccountId, AccountIndex, Address, Balance, BalanceLock, BlockNumber } from '@polkadot/types';
+import { bnMax } from '@polkadot/util';
+
+import { idAndIndex } from '../accounts/idAndIndex';
+import { DerivedBalances } from '../types';
+import { drr } from '../util/drr';
+
+const EMPTY_ACCOUNT = new AccountId(new Uint8Array(32));
+
+export function all (api: ApiInterface$Rx) {
```
Comment (jacogr) [inline packages/api-derive/src/balances/all.ts]: Would like an e2e test here so you can validate it actually does what it says on the tin - https://github.com/polkadot-js/api/blob/master/packages/api/test/e2e/promise-queries.spec.js

File: packages/api-derive/src/balances/all.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { map, switchMap } from 'rxjs/operators';
+import { ApiInterface$Rx } from '@polkadot/api/types';
+import { AccountId, AccountIndex, Address, Balance, BalanceLock, BlockNumber } from '@polkadot/types';
+import { bnMax } from '@polkadot/util';
+
+import { idAndIndex } from '../accounts/idAndIndex';
+import { DerivedBalances } from '../types';
+import { drr } from '../util/drr';
+
+const EMPTY_ACCOUNT = new AccountId(new Uint8Array(32));
```
Comment (jacogr) [inline packages/api-derive/src/balances/all.ts]: Just `new AccountId()` should do the trick.

File: packages/api-derive/src/balances/index.ts (hunk 1 above)
Comment (jacogr) [reply packages/api-derive/src/balances/index.ts]: On second though, let's drop the multi. (We are not using it internally and hopefully the API will have native multi call support real-soon-now - it is my top API priority to do)

**Comment by kwingram25** [comment]
@Tbaut It's "all" as in all derived balance figures, not "all" DOTs/units/etc :)

File: packages/api-derive/src/balances/all.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          // get the maximum of the locks according to https://github.com/paritytech/substrate/blob/master/srml/balances/src/lib.rs#L699
+          lockedBalance = bnMax(...totals.map(({ amount }) => amount)) as Balance;
+        }
+
+        const { offset, perBlock } = vesting.unwrapOr({
+          offset: new Balance(0),
+          perBlock: new Balance(0)
+        } as VestingSchedule);
+
+        const vestedBalance = new Balance(perBlock.mul(bestNumber).add(freeBalance.sub(offset)));
```
Comment (Tbaut) [inline packages/api-derive/src/balances/all.ts]: This can end up being negative, so I think it would make sense to make it 0 in that case.

**Pull request merged**